services/multitenant/tenantstats.go

```diff
@@ -36,6 +73,7 @@ type MultiTenantI interface {
 	RemoveFromInMemoryCount(workspaceID string, destinationType string, count int, tableType string)
 	ReportProcLoopAddStats(stats map[string]map[string]int, tableType string)
 	UpdateWorkspaceLatencyMap(destType string, workspaceID string, val float64)
+	Status() map[string]map[string]map[string]int
```

Comment:
Looks like this map is calling for a composite key, since we are not really interested in actually accessing intermediate levels of the nested maps.
E.g.
```
type MultiTenantStatusKey struct {
	Dataset         string
	Workspace       string
	DestinationType string
}
...
Status() map[Key]int {

```